reactioncommerce/reaction-next-starterkit
# feat: Add Segment tracking for Product List Viewed and Product Clicked events

Issue: #56
Impact: **minor**
Type: **feat**

## Issue

Add event tracking for `Product List Viewed` and `Product Clicked` event for the homepage and tag pages.

## Solution

- Add docs outlining data mappings
- Update `lib/tracking/trackProductListViewed.js` to include outlined data mappings
- Add HOC for tracking product clicked event
- Ensure the `Product List Viewed` is triggered when viewing the homepage and tag pages

## Testing

1. Create a segment account and get the API key. Add the key to the `.env` (see `.env.example`)
2. Start the app navigate to the Product grid homepage and tag pages
3. In the segment dashboard under `sources > javascript > debugger` see the event being tracked with proper data mappings

You should see a `Product List Viewed`, and `Product Clicked` events  something like the following examples:

- https://segment.com/docs/spec/ecommerce/v2/#product-list-viewed 
- https://segment.com/docs/spec/ecommerce/v2/#product-clicked

__Note: The mappings are not going to be 100% the same as segments suggestions.__
